src/renderer/atlas/BackendD3D.cpp

```diff
@@ -447,8 +453,6 @@ void BackendD3D::_recreateCustomShader(const RenderingPayload& p)
     else if (p.s->misc->useRetroTerminalEffect)
     {
         THROW_IF_FAILED(p.device->CreatePixelShader(&custom_shader_ps[0], sizeof(custom_shader_ps), nullptr, _customPixelShader.put()));
-        // We know the built-in retro shader doesn't require continuous redraw.
-        _requiresContinuousRedraw = false;

```

Comment:
This will result in continuous redraw when you go from custom-shader-with-cbuffer-time to retro-shader-builtin, because this is never reset.